### PR TITLE
Cle/5750 nte segment

### PR DESCRIPTION
**Pull request by clediggins-usds**
This PR adds FHIR -> HL7 translation for NTE elements.

Test Steps:
1. Convert a FHIR document to HL7, verify PID, OBX, and OBR NTE segments.

## Changes
- add note.yml
- update observation, observation result, and patient to contain notes mapping
- fix fhir generation that was adding additional spaces into `note` elements.
- add custom `split` function for use in fhirpath

### CODE REVIEW

File: prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt [hunk 1]
```diff
@@ -270,6 +271,25 @@ object CustomFHIRFunctions {
         }
     }
 
+    /**
+     * Splits the [focus] into multiple strings using the delimeter provided in [parameters]
+     */
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt:276]: Add @ return for kdoc

File: prime-router/src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.hl7 [hunk 2]
```diff
@@ -2,6 +2,8 @@ MSH|^~\&#|MEDITECH^2.16.840.1.114222.4.3.2.2.1.321.111^ISO|COCAA^1.2.3.4.5^ISO|D
 SFT|MEDITECH, Inc.^L^^^^MEDITECH&1.3.6.1.4.1.24310&ISO^XX^^^MEDITECH|5.67|Laboratory Applicati|20210810||20210810
```
Comment (carlosfelix2) [inline prime-router/src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.hl7:2]: Do not update this file.  This is the original HL7.

**Review by carlosfelix2 — CHANGES_REQUESTED** [review]

File: prime-router/metadata/hl7_mapping/ORU_R01/base/note.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  - name: note-comment-type
+    value: [ '''C''' ]
+    hl7Spec: [ '%{hl7NotePath}/NTE(%{noteIndex})-4-4' ]
+
+  - name: note-comment-type
+    value: [ '''Comment''' ]
+    hl7Spec: [ '%{hl7NotePath}/NTE(%{noteIndex})-4-5' ]
+
+  - name: note-comment-type
```
Comment (TurkeyFried) [inline prime-router/metadata/hl7_mapping/ORU_R01/base/note.yml:35]: can you change the names in this part to be different for each? makes it easier to search the trace

Comment (clediggins-usds) [reply prime-router/metadata/hl7_mapping/ORU_R01/base/note.yml]: Updated

File: prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt (hunk 1 above)
Comment (clediggins-usds) [reply prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt:277]: updated

File: prime-router/src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.hl7 (hunk 2 above)
Comment (clediggins-usds) [reply prime-router/src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.hl7]: reverted

**Review by carlosfelix2 — APPROVED** [review]

**Review by carlosfelix2 — APPROVED** [review]

**Review by TurkeyFried — APPROVED** [review]

**Pull request merged**